captain/utils/test_sequencer/run_test_sequence.py

```diff
@@ -138,15 +184,20 @@ async def _case_root(node: TestRootNode, **kwargs) -> Extract:
 async def _case_test(node: TestNode, **kwargs) -> Extract:
     # TODO: support run in parallel feature
     await _stream_result_to_frontend(MsgState.RUNNING, test_id=node.id)
-    result, time_taken = _run_pytest(node.path)
+    map_to_runner = {
```

Comment:
you can do this from the `map_to_handler_run` function: 

`map_to_handler_run = (
    "type",
    {
        "root": (None, _case_root),
        "test": (None, _case_test),
        "conditional": (
            "conditional_type",
            {
                "if": (None, _case_if_node),
            },
        ),
    },
)`

since the map_to_handler_run method has a tuple : (key, dict) where key is the field it will search on the element it has, and dict is the mapping to the handler. In your case, you can add "test_type" and then the dict you have here, then split _case_test into -> _case_test_pytest and _case_test_python. Cleaner since to understand, dev has to read only one dict mapping (map_to_hander_run)

Reply:
Already Type Union. Syed and Joey have strong preference for them

Reply:
> But at the same time I'm duplicating the `_case_test` function :/ Which encapsulate the logic that is the same

Good point, but my "vision" was that the developer would need to create a case only for one type of new test or conditional and just add it to the `map_to_handler_run`. Also, here we breakdown the case for tests in the `_case_test` function, but for the conditional we have the specifics in the `map_to_handler` which is also a small inconsistency. 

So, what I propose as a solution is a functional approach: we wrap whatever is a test in a `handle_test` function in the `map_to_handler` function, the wrapper will basically implement the whole broadcasting process! 

I will handle the code for that, no need for you to make a PR :)